Public/Src/FrontEnd/UnitTests/Script.Interpretation/InterpretBinaryExpressions.cs

```diff
@@ -116,7 +116,7 @@ public void PathAndStringAdd()
             Assert.True(((string)result).EndsWith("/myfile.txt`"));
         }
 
-        [FactIfSupported(requiresWindowsBasedOperatingSystem: true)]
+        [Fact]

```

Comment:
Right now, this unit test will test that our `test_runner.sh` script is correct (instead of testing BuildXL).  So for example, if someone runs this test from a debugger or something, the test will fail. 

I don't know how to fix this.  Setting `COMPlus_DefaultStackSize` from the `bxl` process is probably already too late.

In any case, add a comment here saying that this env variable needs to be set to something like 400000 for this test to pass

Reply:
As you've correctly pointed out this has to happen before the CoreCLR initialization - we could do some hacky bxl restart logic but I guess its not worth it. The error happens once DScript gets to a certain complexity level and the type checker recurses to deep, hitting the default stack size limits. I'll add a comment, so we don't forget.